internal/model/vpnoptions_test.go

```diff
@@ -0,0 +1,829 @@
+			t.Errorf("expected error with http uri")
+		}
+	})
+
+}
+
+func Test_parseProto(t *testing.T) {
+	t.Run("fail with empty array of strings", func(t *testing.T) {
+		_, err := parseProto([]string{}, &OpenVPNOptions{})
+		wantErr := ErrBadConfig
+		if !errors.Is(err, wantErr) {
+			t.Errorf("parseProto(): wantErr: %v, got %v", wantErr, err)
+		}
+	})
+
+	t.Run("two parts should fail", func(t *testing.T) {
+		_, err := parseProto([]string{"foo", "bar"}, &OpenVPNOptions{})
+		wantErr := ErrBadConfig
+		if !errors.Is(err, wantErr) {
+			t.Errorf("parseProto(): wantErr %v, got %v", wantErr, err)
+		}
+	})
+
+	t.Run("proto udp is parsed as udp", func(t *testing.T) {
+		opt := &OpenVPNOptions{}
+		o, err := parseProto([]string{"udp"}, opt)
+		if !errors.Is(err, nil) {
+			t.Errorf("parseProto(): wantErr: %v, got %v", nil, err)
+		}
+		if o.Proto != ProtoUDP {
+			t.Errorf("parseProto(): wantErr %v, got %v", nil, err)
+		}
+	})
+
+	t.Run("proto tcp is parsed as tcp", func(t *testing.T) {
+		opt := &OpenVPNOptions{}
+		o, err := parseProto([]string{"tcp"}, opt)
+		if !errors.Is(err, nil) {
+			t.Errorf("parseProto(): wantErr: %v, got %v", nil, err)
+		}
+		if o.Proto != ProtoTCP {
+			t.Errorf("parseProto(): wantErr %v, got %v", nil, err)
+		}
+	})
+
+	t.Run("unknown proto fails", func(t *testing.T) {
+		opt := &OpenVPNOptions{}
+		_, err := parseProto([]string{"kcp"}, opt)
+		wantErr := ErrBadConfig
+		if !errors.Is(err, wantErr) {
+			t.Errorf("parseProto(): wantErr: %v, got %v", ErrBadConfig, err)
+		}
+	})
+}
+
+func Test_parseProxyOBFS4(t *testing.T) {
+	t.Run("with empty parts", func(t *testing.T) {
+		_, err := parseProxyOBFS4([]string{}, &OpenVPNOptions{})
+		wantErr := ErrBadConfig
+		if !errors.Is(err, wantErr) {
+			t.Errorf("parseProxyOBFS4(): wantErr: %v, got %v", wantErr, err)
+		}
+	})
+
+	t.Run("with an obfs4 string", func(t *testing.T) {
+		// TODO(ainghazal): this test must change when the function starts validating the obfs4 url
+		opt := &OpenVPNOptions{}
+		obfs4Uri := "obfs4://foobar"
+		o, err := parseProxyOBFS4([]string{obfs4Uri}, opt)
+		var wantErr error = nil
+		if !errors.Is(err, wantErr) {
+			t.Errorf("parseProxyOBFS4(): wantErr: %v, got %v", wantErr, err)
+		}
+		if o.ProxyOBFS4 != obfs4Uri {
+			t.Errorf("parseProxyOBFS4(): want %v, got %v", obfs4Uri, opt.ProxyOBFS4)
+		}
+	})
+}
+
+func Test_parseCA(t *testing.T) {
+	t.Run("more than one part should fail", func(t *testing.T) {
+		_, err := parseCA([]string{"one", "two"}, &OpenVPNOptions{}, "")
+		wantErr := ErrBadConfig
+		if !errors.Is(err, wantErr) {
+			t.Errorf("parseCA(): want %v, got %v", wantErr, err)
+		}
+	})
+
+	t.Run("empty part should fail", func(t *testing.T) {
+		_, err := parseCA([]string{}, &OpenVPNOptions{}, "")
+		var wantErr error = ErrBadConfig
+		if !errors.Is(err, wantErr) {
+			t.Errorf("parseCA(): want %v, got %v", wantErr, err)
+		}
+	})
+}
+
+func Test_parseCert(t *testing.T) {
+	t.Run("more than one part should ", func(t *testing.T) {
```

Comment:
I think this test string is missing some extra test:

```suggestion
	t.Run("more than one part should fail", func(t *testing.T) {
```